core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -423,22 +421,13 @@ public int parseInto(ReadWritablePeriod period, String text, int position, Local
                     }
                 }
                 else if (parsePos < 0) {
-                    parsePos = ~parsePos;
-                    if (parsePos > bestInvalidPos) {
-                        bestInvalidPos = parsePos;
-                    }
-                }
-            }
-
-            if (bestValidPos > position || (bestValidPos == position)) {
-                // Restore the state to the best valid parse.
-                if (bestValidPeriod != null) {
-                    period.setPeriod(bestValidPeriod);
+                    return ~parsePos;
                 }
-                return bestValidPos;
             }
 
-            return ~bestInvalidPos;
```

Comment:
This line is unreachable because `bestValidPos > position || (bestValidPos == position)` is always true.